pkg/apis/pipeline/v1/result_defaults.go

```diff
@@ -13,13 +13,22 @@ limitations under the License.
 
 package v1
 
-import "context"
+import (
+	"context"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+)
 
 // SetDefaults set the default type for TaskResult
-func (tr *TaskResult) SetDefaults(context.Context) {
+func (tr *TaskResult) SetDefaults(ctx context.Context) {
 	if tr == nil {
 		return
 	}
+	if tr.Value == nil && config.FromContextOrDefaults(ctx).FeatureFlags.EnableStepActions {
+		tr.Value = &ParamValue{
+			Type: ParamTypeString,
+		}
+	}
```

Comment:
what if the TaskResult.Type is not string?

Reply:
Good question. I actually removed defaults since it is an optional field. We can just take care of it based on if the value is `nil`. No need to assign it a default. 

It was also breaking some other e2e tests so it might cause broken behaviour. I got rid of it.

Reply:
yeah, that makes sense, resolved